src/main/groovy/io/github/groovymc/modsdotgroovy/AbstractConvertTask.groovy

```diff
@@ -0,0 +1,149 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2022 GroovyMC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.github.groovymc.modsdotgroovy
+
+import groovy.transform.CompileDynamic
+import groovy.transform.CompileStatic
+import org.codehaus.groovy.control.CompilerConfiguration
+import org.gradle.api.DefaultTask
+import org.gradle.api.artifacts.Configuration
+import org.gradle.api.file.CopySpec
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.provider.MapProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.*
+import org.gradle.language.jvm.tasks.ProcessResources
+
+import java.nio.file.Files
+
+abstract class AbstractConvertTask extends DefaultTask {
+    @InputFile
+    abstract RegularFileProperty getInput()
+    @Optional
+    @OutputFile
+    abstract RegularFileProperty getOutput()
+    @Optional
+    @InputFile
+    abstract RegularFileProperty getDslLocation()
+    @Optional
+    @Input
+    abstract Property<Configuration> getDslConfiguration()
```

Comment:
Not sure what to say about this... Since you can just `afterEvaluate {dslLocation = dslConfiguration.resolve().find()}`
But I guess it can stay.

Reply:
That's a good point. It may be work keeping for backwards compat, but I could stick an `@Deprecated` on it